lib/async_hooks.js

```diff
@@ -61,13 +61,7 @@ async_wrap.setupHooks({ init,
 
 // Used to fatally abort the process if a callback throws.
 function fatalError(e) {
-  if (typeof e.stack === 'string') {
-    process._rawDebug(e.stack);
-  } else {
-    const o = { message: e };
-    Error.captureStackTrace(o, fatalError);
-    process._rawDebug(o.stack);
-  }
+  process._rawDebug(e.stack);
```

Comment:
So... We might want this exposed as the standard way to fail from a call back or `Promise.catch`... so failing with just a message might be useful... Also we're trying to remove non `NodeError`s from the code (even if they are just stack containers).
IMHO restore the weird `captureStackTrace` case, and please add it to the `module.exports = {`
Ref: https://github.com/nodejs/node/pull/13839#discussion_r124936232